languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -88,6 +88,49 @@ def self.join(separator, left, right)
         ffi_polar.validate_roles_config(validation_query_results)
       end
 
+      def get_allowed_resources(actor, action, klass) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        resource = Variable.new 'resource'
+
+        # FIXME(gw) kind of a hack because of class autoreloading, maybe not needed?
+        if host.types.key? klass
+          class_name = host.types[klass].name
+        elsif host.types.key? klass.name
+          class_name = host.types[klass.name].name
+        else
+          raise "Unknown class `#{klass}`"
+        end
+
+        constraint = Expression.new(
+          'And',
+          [Expression.new('Isa', [resource, Pattern.new(class_name, {})])]
+        )
+
+        results = query_rule(
+          'allow',
+          actor,
+          action,
+          resource,
+          bindings: { 'resource' => constraint },
+          accept_expression: true
+        )
+
+        complete = []
+        partial = []
+
+        results.to_a.each do |result|
```

Comment:
Nit alert: do you need `to_a` here?

Reply:
yes there does, otherwise it complains about calling a private method, go fig